### ISSUE TITLE
proc newVariant()

### ISSUE THREAD

**Issue by StefanSalewski** [StefanSalewski/gintro#178]
```nim
proc newVariantBoolean*(value: bool): Variant =
  fnew(result, finalizerunref)
  result.impl = g_variant_new_boolean(gboolean(value))
  result.ignoreFinalizer = true
```

https://docs.gtk.org/gio/ctor.SimpleAction.new_stateful.html

"If the state GVariant is floating, it is consumed."

We should check that. GVariant is not a GObject , but can be floating.

**Comment by StefanSalewski** [comment]
See

https://discourse.gnome.org/t/g-variant-new-and-transfer-none/2472

**Comment by StefanSalewski** [comment]
So something like this may be good enough?

```nim
proc newVariantBoolean*(value: bool): Variant =
  fnew(result, finalizerunref)
  result.impl = g_variant_new_boolean(gboolean(value))
  assert g_variant_is_floating(result.impl).int == GTrue.int
  #result.ignoreFinalizer = true
  discard g_variant_ref_sink(result.impl)
  assert g_variant_is_floating(result.impl).int == GFalse.int
```

**Comment by StefanSalewski** [comment]
Should be fixed for v0.9.5
